python/lsst/ts/envsensors/sensor/base_sensor.py

```diff
@@ -0,0 +1,67 @@
+# This file is part of ts_envsensors.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["BaseSensor", "DELIMITER", "TERMINATOR"]
+
+from abc import ABC, abstractmethod
+import logging
+from typing import List, Union
+
+"""Serial data line terminator."""
+TERMINATOR = "\r\n"
+
+"""Serial data channel delimiter."""
+DELIMITER = ","
+
+
+class BaseSensor(ABC):
+    """Base class for the different types of Sensors.
+
+    This class holds all common code for the sensors. Sensor specific code
+    (for instance temperature or wind sensor) needs to be implemented in a
+    sub-class.
+
+    Parameters:
+    -----------
+    channels: `int`
+        The number of channels that the sensor will produce telemetry for.
+    log: `logging.Logger`
+        The logger to create a child logger for.
+    """
+
+    @abstractmethod
+    def __init__(self, channels: int, log: logging.Logger) -> None:
+        self.channels: int = channels
+        self._log = log.getChild(type(self).__name__)
+        self.terminator: str = TERMINATOR
```

Comment:
If terminator is subclass-specific then I strongly suggest making this a class variable (define it before `__init__` without the `self.`). Then subclasses can easily override it and it will be shared by all instances of the given class.

If it is instance-specific then please make it a constructor argument (presumably with default value TERMINATOR).

Reply:
I removed terminator since it is not used anywhere.